src/Transform/ONNX/Decompose.cpp

```diff
@@ -747,6 +747,138 @@ struct CustomOpFuseMatMulPattern : public OpRewritePattern<ONNXCustomOp> {
   }
 };
 
+// Transform InstanceNormalization into LayerNormalization
+struct InstanceNormIntoLayerNormPattern
+    : public OpRewritePattern<ONNXInstanceNormalizationOp> {
+  using OpRewritePattern<ONNXInstanceNormalizationOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(ONNXInstanceNormalizationOp instanceNormOp,
+      PatternRewriter &rewriter) const final {
+    // Match.
+    Value input = instanceNormOp.getInput();
+    if (!input.getType().isa<ShapedType>())
+      return failure();
+
+    // Get info.
+    Value scale = instanceNormOp.getScale();
+    Value bias = instanceNormOp.getB();
+    ShapedType inputType = input.getType().cast<ShapedType>();
+    Type elementType = inputType.getElementType();
+    auto inputShape = inputType.getShape();
+    int64_t C = inputShape[1];
+    int64_t inputRank = inputType.getRank();
+    assert(inputRank > 2 && "expected instance norm with input ranks > 2");
```

Comment:
conceptually, is the "magic number" 2 here the same as the 2 in axis in line 775? consider naming the constant 2 in a way so that you can refer to it by name, both in InstanceNormIntoLayerNormPattern and GroupNormIntoLayerNormPattern where the number 2 appears multiple times too

Reply:
good idea.